packages/core/src/common/disposable.ts

```diff
@@ -28,13 +28,19 @@ export namespace Disposable {
         return !!arg && typeof arg === 'object' && 'dispose' in arg && typeof arg['dispose'] === 'function';
     }
     export function create(func: () => void): Disposable {
-        return {
-            dispose: func
-        };
+        return { dispose: func };
     }
-    export const NULL = create(() => { });
+    export declare const NULL: Disposable;
 }
 
+Object.defineProperty(Disposable, 'NULL', {
```

Comment:
I would add a comment for the next person reading this oddity:

```suggestion
// There are issues with sharing the same disposable reference around, so we
// need to make `Disposable.NULL` return new references for each access.
Object.defineProperty(Disposable, 'NULL', {
```